packages/core/src/browser/drag-and-drop.ts

```diff
@@ -0,0 +1,137 @@
+/*
+* Copyright (C) 2017 Ericsson and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from "inversify";
+import { TreeWidget, TreeProps, ITreeModel, ITreeNode, NodeProps } from './tree';
+import { ContextMenuRenderer } from "./context-menu-renderer";
+import { ElementAttrs } from "@phosphor/virtualdom";
+
+export const DROPZONE_CLASS = 'dropzone';
+
+@injectable()
+export class DragAndDraggable extends TreeWidget {
+
+    constructor(
+        @inject(TreeProps) readonly props: TreeProps,
+        @inject(ITreeModel) readonly model: ITreeModel,
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer
+    ) {
+        super(props, model, contextMenuRenderer);
+    }
+
+    protected createNodeAttributes(node: ITreeNode, props: NodeProps): ElementAttrs {
+        const elementAttrs: ElementAttrs = super.createNodeAttributes(node, props);
+        const childExpansion = this.isExandable(node) ? 0 : 1;
+
+        if (!childExpansion) {
+            // Folder, so it could be a dropzone
+            return {
+                ...elementAttrs,
+                ondragenter: event => this.handleDragEnterEvent(node, event),
+                ondrop: event => this.handleDropEvent(node, event),
+                ondragover: event => this.handleDragOverEvent(node, event),
+                ondragend: event => console.log("END"),
```

Comment:
`consol.log` is temporary right? It should be fixed in all places in the PR.

Reply:
OK , 
console.log()  are temporary, Here for the events "ondragend" and "ondragexit", those events are not called for the external file, but those events , I expect to be called when handling the files inside THEIA. I just removed the events for now.

Reply:
In File-tree-widget, next version, on console.log replace by console.error()

Reply:
Please use the Logger, logger.error()